specs/governance/gov-token.md

```diff
@@ -0,0 +1,93 @@
+from the `Ownable` contract. When tokens are minted, the voting power of the recipient address MUST be updated accordingly.
+
+### Token Burning
+
+The contract MUST allow token holders to burn their own tokens using the inherited `burn` function from `ERC20Burnable`.
+When tokens are burned, the total supply and the holder's voting power MUST be reduced accordingly.
+
+### Voting Power
+
+Each token corresponds to one unit of voting power.
+By default, token balance does not account for voting power. To have their voting power counted, token holders MUST delegate
+their voting power to an address (can be their own address).
+The contract MUST offer public accessors for quering voting power, as outlined below.
+
+#### Queries
+
+- The `getVotes` function MUST return the current voting power of an address.
+- The `getPastVotes` function MUST allow querying the voting power of an address at a specific block number in the past.
+- The `getPastTotalSupply` function MUST return the total voting power at a specific block number in the past.
+
+### Delegation
+
+Vote power can be delegated either by calling the `delegate` function directly or by providing a signature to be used
+with `delegateBySig`, as inherited from `ERC20Votes`.
+
+When a token holder delegates their voting power, the delegated address receives the voting power corresponding to the token
+holder's balance.
+
+Delegation is recorded in checkpoints, which store the voting power of each address at specific points in time.
+
+### Checkpoints
+
+The contract keeps a history of each account's voting power using checkpoints.
+Checkpoints are recorded whenever tokens are transferred, minted, or burned, and when delegation occurs.
+Each checkpoint represents the voting power of an address at a specific block number.
+The contract MUST maintain a mapping of addresses to their checkpoint history. Additionally, the contract MUST
+provide functions `numCheckpoints` and `checkpoints` to allow retrieving the number of checkpoints for an account
+and the details of a specific checkpoint.
+
+### Overrides
+
+The contract overrides several functions from the inherited contracts to prevent collisions between the inherited contracts
+methods. The following functions are overridden:
+
+- `_afterTokenTransfer`: Called after each token transfer to update the voting power. Concretely, the contract uses
+  the `ERC20Votes` implementation to override the `_afterTokenTransfer` function.
+- `_mint`: Called when new tokens are minted to update the voting power and total supply. The contract uses the
+  `ERC20Votes` implementation to override the `_mint` function.
+- `_burn`: Called when tokens are burned to update the voting power and total supply. The contract uses the `ERC20Votes`
+  implementation to override the `_burn` function.
+
+### Supply Cap
+
+The total token supply is capped to `2^208^ - 1` to prevent overflow risks in the voting system.
```

Comment:
We are on an older version of the oz contracts before they added custom errors

Reply:
I just checked and our current version of oz contracts seems to include custom errors, at least in `Governor`. could you elaborate more?

Reply:
the current [GovernanceToken](https://github.com/voteagora/governor/blob/main/src/lib/OptimismToken.sol) implementation uses v4.5.0, which doesn't use/include custom errors.

Reply:
ok cool. that makes sense